### PR TITLE
remove a Py_XDECREF on a ref we lost ownership of

### PR DESCRIPTION
**Pull request by arbll**
### What does this PR do?

https://docs.python.org/2/extending/extending.html#ownership-rules
> PyTuple_SetItem() and PyList_SetItem(). These functions take over ownership of the item passed to them — even if they fail!

and https://github.com/python/cpython/blob/2.7/Doc/c-api/intro.rst#reference-count-details

https://github.com/DataDog/datadog-agent/blob/6f642b5daf9b35d755b31db853c8346794c48dbc/six/two/two.cpp#L288
https://github.com/DataDog/datadog-agent/blob/6f642b5daf9b35d755b31db853c8346794c48dbc/six/two/two.cpp#L344

### CODE REVIEW

File: six/two/two.cpp [hunk 1]
```diff
@@ -341,7 +341,6 @@ bool Two::getCheck(SixPyObject *py_class, const char *init_config_str, const cha
     Py_XDECREF(name);
     Py_XDECREF(check_id);
     Py_XDECREF(init_config);
-    Py_XDECREF(instance);
```
Comment (olivielpeau) [inline six/two/two.cpp]: This deserves one or more comments on why we intentionally don't decref `instance`.

I'm thinking:
* one comment at the beginning of the `done` block
* one comment when we use `PyTuple_SetItem` to say explicitly that the reference is stolen

**Review by olivielpeau — DISMISSED** [review]
Great catch!

Definitely deserves a couple of comments in the code.

File: six/two/two.cpp (hunk 1 above)
Comment (truthbk) [reply six/two/two.cpp]: I agree we should definitely better document this. 

Wondering if it wouldn't be better to use `PyObject_SetItem()` instead for a more consistent developer experience in the future? Although maybe having this explicitly documented is less likely to re-introduce this tricky bug in the future. If we do change this to use `PyObject_SetItem()`, we should add a big fat comment too. 

I'm fine with merging this as-is once the comment is in place.

Comment (truthbk) [reply six/two/two.cpp]: @arbll go ahead and merge it once you add the nice big fat comments. If we decide to refactor some of this we can do it in a later PR.

**Review by olivielpeau — DISMISSED** [review]

**Pull request merged**